<a href="https://colab.research.google.com/github/bysubanji/Health_insurance_cross_sell_prediction/blob/main/Health_Insurance_Bhaskarsubanji.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Cross Sell Prediction of Health Insurance**

### **Predict Health Insurance Owners' who will be interested in Vehicle Insurance**

Your client is an Insurance company that has supplied Health Insurance to its customers now they require your help in building a model to predict whether the consumers from the past year will also be interested in Vehicle Insurance provided by the company.

**Objective**

Building a model to predict whether a customer would be interested in Vehicle Insurance is extremely helpful for the company because it can then accordingly plan its communication strategy to reach out to those customers and optimise its business model and revenue.

# Problem and Data importing

## Importing Required Libreries, mounting of drive and loading the dataset.

In [1]:
# importing libraries
import pandas  as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.preprocessing import LabelEncoder 
from sklearn.model_selection import train_test_split
from imblearn.over_sampling import RandomOverSampler
from collections import Counter
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import precision_score,recall_score,accuracy_score,f1_score,confusion_matrix,roc_auc_score,classification_report
from sklearn.ensemble import RandomForestClassifier
from xgboost import XGBClassifier

In [2]:
# Mounting of Drive
from google.colab import drive
drive.mount("/content/drive")

Mounted at /content/drive


In [3]:
# Loading of data
data=pd.read_csv("/content/drive/MyDrive/Projects - Almabetter/Health_Insurance_cross_sell_production/TRAIN-HEALTH INSURANCE CROSS SELL PREDICTION.csv")

# Data summarization

In [4]:
#showing first 5 rows
data.head()

id  Gender  Age  ...  Policy_Sales_Channel  Vintage  Response
0   1    Male   44  ...                  26.0      217         1
1   2    Male   76  ...                  26.0      183         0
2   3    Male   47  ...                  26.0       27         1
3   4    Male   21  ...                 152.0      203         0
4   5  Female   29  ...                 152.0       39         0

[5 rows x 12 columns]

In [5]:
# showing last 5 rows
data.tail()

id  Gender  Age  ...  Policy_Sales_Channel  Vintage  Response
381104  381105    Male   74  ...                  26.0       88         0
381105  381106    Male   30  ...                 152.0      131         0
381106  381107    Male   21  ...                 160.0      161         0
381107  381108  Female   68  ...                 124.0       74         0
381108  381109    Male   46  ...                  26.0      237         0

[5 rows x 12 columns]

In [6]:
# check datatypes, shape,null values
print(data.info())
print()
print(data.isna().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 381109 entries, 0 to 381108
Data columns (total 12 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   id                    381109 non-null  int64  
 1   Gender                381109 non-null  object 
 2   Age                   381109 non-null  int64  
 3   Driving_License       381109 non-null  int64  
 4   Region_Code           381109 non-null  float64
 5   Previously_Insured    381109 non-null  int64  
 6   Vehicle_Age           381109 non-null  object 
 7   Vehicle_Damage        381109 non-null  object 
 8   Annual_Premium        381109 non-null  float64
 9   Policy_Sales_Channel  381109 non-null  float64
 10  Vintage               381109 non-null  int64  
 11  Response              381109 non-null  int64  
dtypes: float64(3), int64(6), object(3)
memory usage: 34.9+ MB
None

id                      0
Gender                  0
Age                     0
Driving_Licens

Dataset has 381109 rows and 12 columns. There is no missing values present in dataset.

In [7]:
#Statistical Summary of Numeric Variables
data.describe()

id            Age  ...        Vintage       Response
count  381109.000000  381109.000000  ...  381109.000000  381109.000000
mean   190555.000000      38.822584  ...     154.347397       0.122563
std    110016.836208      15.511611  ...      83.671304       0.327936
min         1.000000      20.000000  ...      10.000000       0.000000
25%     95278.000000      25.000000  ...      82.000000       0.000000
50%    190555.000000      36.000000  ...     154.000000       0.000000
75%    285832.000000      49.000000  ...     227.000000       0.000000
max    381109.000000      85.000000  ...     299.000000       1.000000

[8 rows x 9 columns]